gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -30,34 +30,56 @@ var fluid = fluid || require("infusion"),
     /**
      * APP_SETTING_PRIORITY: the priority that an application is given if the user has application
      *    specific settings in their NP set for that application, but has no explicit priority for
-     *    that application.
-     * MIN_USER_PRIORITY: If an explicit priority is set by the user for an application, this is
-     *    the minimum value it should have, to avoid conflicting with system generated priorities.
+     *    that application. * MIN_USER_PRIORITY: If an explicit priority is set by the user for an application, this is *    the minimum value it should have, to avoid conflicting with system generated priorities.
      */
 
     gpii.matchMakerFramework.APP_SETTING_PRIORITY = 512;
     gpii.matchMakerFramework.MIN_USER_PRIORITY = 1024;
 
+    /*
+     * responsible for building the input payload to the matchmaker, via a bunch of helper functions
+     * initialPayload consists of:
```

Comment:
Comment should be clear on whether the input argument is modified since it isn't totally clear from the implementation

Reply:
good point! It would depend on the specific matchmaker implementation whether the input is modified - and while I dont think neither the canopy MM nor flat MM touches the inputs right now, I've changed the code to make a copy of it and pass that on - though it goes against all of Kasparnets principles not to leave a time bomb bug like that laying around :)